diff -urN aa/gnome-system-monitor-2.5.3/configure.in gnome-system-monitor-2.5.3/configure.in
--- aa/gnome-system-monitor-2.5.3/configure.in	2004-02-15 20:16:09.000000000 +0100
+++ gnome-system-monitor-2.5.3/configure.in	2004-02-15 22:19:58.468749680 +0100
@@ -29,7 +29,7 @@
 AC_DEFINE_UNQUOTED(GNOME_ICONDIR, "${prefix}/share/pixmaps")
 
 dnl Add the languages which your application supports here.
-ALL_LINGUAS="am ar az be bg bn ca cs cy da de el es fa fi fr ga gl he hi hr hu id it ja ko li lt lv mk ml mn ms nl no pl pt pt_BR ro ru sk sl sq sr sr@Latn sv ta tr uk vi zh_CN zh_TW"
+ALL_LINGUAS="am ar az be bg bn ca cs cy da de el es fa fi fr ga gl he hi hr hu id it ja ko li lt lv mk ml mn ms nb nl pl pt pt_BR ro ru sk sl sq sr sr@Latn sv ta tr uk vi zh_CN zh_TW"
 
 GETTEXT_PACKAGE=gnome-system-monitor
 AC_SUBST(GETTEXT_PACKAGE)
