--- gnome-system-monitor-2.5.0/src/interface.c.wiget	2003-11-12 21:21:38.000000000 +0100
+++ gnome-system-monitor-2.5.0/src/interface.c	2003-11-12 21:48:05.000000000 +0100
@@ -383,12 +383,8 @@
 	hbox = gtk_hbox_new (FALSE, 12);
 	gtk_box_pack_start (GTK_BOX (cpu_graph_box), hbox, FALSE, FALSE, 0);
 
-	color_picker = gnome_color_picker_new ();
-	gnome_color_picker_set_i16 (GNOME_COLOR_PICKER (color_picker), 
-				    cpu_graph->colors[2].red,
-				    cpu_graph->colors[2].green,
-				    cpu_graph->colors[2].blue, 0);
-	g_signal_connect (G_OBJECT (color_picker), "color_set",
+	color_picker = gtk_color_button_new_with_color (cpu_graph->colors);
+	g_signal_connect (G_OBJECT (color_picker), "color-set",
 			    G_CALLBACK (cb_cpu_color_changed), procdata);
 	gtk_box_pack_start (GTK_BOX (hbox), color_picker, FALSE, FALSE, 0);
 	
@@ -431,12 +427,8 @@
 	gtk_box_pack_start (GTK_BOX (mem_graph_box), table, 
 			    FALSE, FALSE, 0);
 
-	color_picker = gnome_color_picker_new ();
-	gnome_color_picker_set_i16 (GNOME_COLOR_PICKER (color_picker), 
-				    mem_graph->colors[2].red,
-				    mem_graph->colors[2].green,
-				    mem_graph->colors[2].blue, 0);
-	g_signal_connect (G_OBJECT (color_picker), "color_set",
+	color_picker = gtk_color_button_new_with_color (mem_graph->colors);
+	g_signal_connect (G_OBJECT (color_picker), "color-set",
 			    G_CALLBACK (cb_mem_color_changed), procdata);
 	gtk_table_attach (GTK_TABLE (table), color_picker, 0, 1, 0, 1, 0, 0, 0, 0);
 	
@@ -458,12 +450,8 @@
 	gtk_table_attach (GTK_TABLE (table), mem_graph->memtotal_label, 4, 5, 0, 1, 
 			  GTK_FILL, 0, 0, 0);
 	
-	color_picker = gnome_color_picker_new ();
-	gnome_color_picker_set_i16 (GNOME_COLOR_PICKER (color_picker), 
-				    mem_graph->colors[3].red,
-				    mem_graph->colors[3].green,
-				    mem_graph->colors[3].blue, 0);
-	g_signal_connect (G_OBJECT (color_picker), "color_set",
+	color_picker = gtk_color_button_new_with_color (mem_graph->colors);
+	g_signal_connect (G_OBJECT (color_picker), "color-set",
 			    G_CALLBACK (cb_swap_color_changed), procdata);
 	gtk_table_attach (GTK_TABLE (table), color_picker, 0, 1, 1, 2, 0, 0, 0, 0);
 			  
--- gnome-system-monitor-2.5.0/src/callbacks.h.wiget	2003-11-12 21:26:04.000000000 +0100
+++ gnome-system-monitor-2.5.0/src/callbacks.h	2003-11-12 21:49:35.000000000 +0100
@@ -70,12 +70,9 @@
 void		cb_search (GtkEditable *editable, gpointer data);
 
 
-void		cb_cpu_color_changed (GnomeColorPicker *cp, guint r, guint g, guint b,
-				      guint a, gpointer data);
-void		cb_mem_color_changed (GnomeColorPicker *cp, guint r, guint g, guint b,
-				      guint a, gpointer data);
-void		cb_swap_color_changed (GnomeColorPicker *cp, guint r, guint g, guint b,
-				      guint a, gpointer data);
+void		cb_cpu_color_changed (GtkColorButton *cp, gpointer data);
+void		cb_mem_color_changed (GtkColorButton *cp, gpointer data);
+void		cb_swap_color_changed (GtkColorButton *cp, gpointer data);
 
 void		cb_row_selected (GtkTreeSelection *selection, gpointer data);
 gboolean	cb_tree_button_pressed (GtkWidget *widget, GdkEventButton *event, 
--- gnome-system-monitor-2.5.0/src/callbacks.c.wiget	2003-11-12 21:28:19.000000000 +0100
+++ gnome-system-monitor-2.5.0/src/callbacks.c	2003-11-12 21:53:00.000000000 +0100
@@ -362,40 +362,43 @@
 
 
 void		
-cb_cpu_color_changed (GnomeColorPicker *cp, guint r, guint g, guint b,
-		      guint a, gpointer data)
+cb_cpu_color_changed (GtkColorButton *cp, gpointer data)
 {
 	ProcData *procdata = data;
 	GConfClient *client = procdata->client;
 	gchar *color;
+	GdkColor *g_color;
 	
-	color = g_strdup_printf("#%02x%02x%02x", r, g, b);
+	gtk_color_button_get_color(cp, g_color);
+	color = g_strdup_printf("#%02x%02x%02x", g_color->red, g_color->green, g_color->blue);
 	gconf_client_set_string (client, "/apps/procman/cpu_color", color, NULL);
 	g_free (color);
 }
 
 void		
-cb_mem_color_changed (GnomeColorPicker *cp, guint r, guint g, guint b,
-		      guint a, gpointer data)
+cb_mem_color_changed (GtkColorButton *cp, gpointer data)
 {
 	ProcData *procdata = data;
 	GConfClient *client = procdata->client;
 	gchar *color;
+	GdkColor *g_color;
 
-	color = g_strdup_printf("#%02x%02x%02x", r, g, b);
+	gtk_color_button_get_color(cp, g_color);
+	color = g_strdup_printf("#%02x%02x%02x", g_color->red, g_color->green, g_color->blue);
 	gconf_client_set_string (client, "/apps/procman/mem_color", color, NULL);
 	g_free (color);
 }
 
 void		
-cb_swap_color_changed (GnomeColorPicker *cp, guint r, guint g, guint b,
-		       guint a, gpointer data)
+cb_swap_color_changed (GtkColorButton *cp, gpointer data)
 {
 	ProcData *procdata = data;
 	GConfClient *client = procdata->client;
 	gchar *color;
+	GdkColor *g_color;
 
-	color = g_strdup_printf("#%02x%02x%02x", r, g, b);
+	gtk_color_button_get_color(cp, g_color);
+	color = g_strdup_printf("#%02x%02x%02x", g_color->red, g_color->green, g_color->blue);
 	gconf_client_set_string (client, "/apps/procman/swap_color", color, NULL);
 	g_free (color);
 }
--- gnome-system-monitor-2.5.0/src/Makefile.am.wiget	2003-11-12 22:17:44.000000000 +0100
+++ gnome-system-monitor-2.5.0/src/Makefile.am	2003-11-12 22:18:02.000000000 +0100
@@ -3,7 +3,7 @@
 INCLUDES = \
 	-DPROCMAN_DATADIR=\"$(datadir)/procman/\" \
 	-DDATADIR=\""$(datadir)"\" \
-	@PROCMAN_CFLAGS@ -DGTK_DISABLE_DEPRECATED=1 \
+	@PROCMAN_CFLAGS@ \
 	-DGNOME_DISABLE_DEPRECATED
 
 bin_PROGRAMS = gnome-system-monitor
--- gnome-system-monitor-2.5.0/src/procdialogs.c.wiget	2003-11-12 21:36:36.000000000 +0100
+++ gnome-system-monitor-2.5.0/src/procdialogs.c	2003-11-12 22:00:02.000000000 +0100
@@ -422,27 +422,29 @@
 }
 
 static void		
-bg_color_changed (GnomeColorPicker *cp, guint r, guint g, guint b,
-		  guint a, gpointer data)
+bg_color_changed (GtkColorButton *cp, gpointer data)
 {
 	ProcData *procdata = data;
 	GConfClient *client = procdata->client;
 	gchar *color;
-	
-	color = g_strdup_printf("#%02x%02x%02x", r, g, b);
+	GdkColor *g_color;
+
+	gtk_color_button_get_color(cp, g_color);
+	color = g_strdup_printf("#%02x%02x%02x", g_color->red, g_color->green, g_color->blue);
 	gconf_client_set_string (client, "/apps/procman/bg_color", color, NULL);
 	g_free (color);
 }
 
 static void		
-frame_color_changed (GnomeColorPicker *cp, guint r, guint g, guint b,
-		  guint a, gpointer data)
+frame_color_changed (GtkColorButton *cp, gpointer data)
 {
 	ProcData *procdata = data;
 	GConfClient *client = procdata->client;
 	gchar *color;
-	
-	color = g_strdup_printf("#%02x%02x%02x", r, g, b);
+	GdkColor *g_color;
+
+	gtk_color_button_get_color(cp, g_color);
+	color = g_strdup_printf("#%02x%02x%02x", g_color->red, g_color->green, g_color->blue);
 	gconf_client_set_string (client, "/apps/procman/frame_color", color, NULL);
 	g_free (color);
 }
@@ -725,12 +727,8 @@
 	gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
 	gtk_box_pack_start (GTK_BOX (hbox2), label, FALSE, FALSE, 0);
 	
-	color_picker = gnome_color_picker_new ();
-	gnome_color_picker_set_i16 (GNOME_COLOR_PICKER (color_picker), 
-				    procdata->config.bg_color.red,
-				    procdata->config.bg_color.green,
-				    procdata->config.bg_color.blue, 0);
-	g_signal_connect (G_OBJECT (color_picker), "color_set",
+	color_picker = gtk_color_button_new_with_color (&(procdata->config.bg_color));
+	g_signal_connect (G_OBJECT (color_picker), "color-set",
 			          G_CALLBACK (bg_color_changed), procdata);
 	gtk_box_pack_end (GTK_BOX (hbox2), color_picker, FALSE, FALSE, 0);
 	gtk_label_set_mnemonic_widget (GTK_LABEL (label), color_picker);
@@ -744,12 +742,8 @@
 	gtk_misc_set_alignment (GTK_MISC (label), 0.0, 0.5);
 	gtk_box_pack_start (GTK_BOX (hbox2), label, FALSE, FALSE, 0);
 	
-	color_picker = gnome_color_picker_new ();
-	gnome_color_picker_set_i16 (GNOME_COLOR_PICKER (color_picker), 
-				    procdata->config.frame_color.red,
-				    procdata->config.frame_color.green,
-				    procdata->config.frame_color.blue, 0);
-	g_signal_connect (G_OBJECT (color_picker), "color_set",
+	color_picker = gtk_color_button_new_with_color (&(procdata->config.frame_color));
+	g_signal_connect (G_OBJECT (color_picker), "color-set",
 			    G_CALLBACK (frame_color_changed), procdata);
 	gtk_size_group_add_widget (size, color_picker);	  
 	gtk_box_pack_end (GTK_BOX (hbox2), color_picker, FALSE, FALSE, 0);	
