diff -aurN gnome-system-monitor-2.12.2.orig/gnome-system-monitor.desktop.in.in gnome-system-monitor-2.12.2/gnome-system-monitor.desktop.in.in
--- gnome-system-monitor-2.12.2.orig/gnome-system-monitor.desktop.in.in	2005-11-29 09:32:19.000000000 +0100
+++ gnome-system-monitor-2.12.2/gnome-system-monitor.desktop.in.in	2005-11-29 18:05:02.000000000 +0100
@@ -8,7 +8,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=Application;System;Monitor;X-Red-Hat-Base
+Categories=GTK;GNOME;System;Monitor;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=system-monitor
 X-GNOME-Bugzilla-Component=general
