diff -aurN gnome-system-monitor-2.13.90.orig/gnome-system-monitor.desktop.in.in gnome-system-monitor-2.13.90/gnome-system-monitor.desktop.in.in
--- gnome-system-monitor-2.13.90.orig/gnome-system-monitor.desktop.in.in	2006-01-29 20:05:56.000000000 +0100
+++ gnome-system-monitor-2.13.90/gnome-system-monitor.desktop.in.in	2006-02-01 21:53:23.361476000 +0100
@@ -8,7 +8,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Application;System;Monitor;X-Red-Hat-Base;
+Categories=GTK;GNOME;System;Monitor;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=system-monitor
 X-GNOME-Bugzilla-Component=general
